pandas/core/arrays/datetimes.py

```diff
@@ -108,3 +109,22 @@ def _sub_datelike_dti(self, other):
             mask = (self._isnan) | (other._isnan)
             new_values[mask] = iNaT
         return new_values.view('timedelta64[ns]')
+
+    # -----------------------------------------------------------------
+    # Timezone Conversion and Localization Methods
+
+    def _local_timestamps(self):
+        """
+        Convert to an i8 (unix-like nanosecond timestamp) representation
+        while keeping the local timezone and not using UTC.
+        This is used to calculate time-of-day information as if the timestamps
+        were timezone-naive.
+        """
+        values = self.asi8
+        indexer = values.argsort()
+        result = conversion.tz_convert(values.take(indexer), utc, self.tz)
```

Comment:
Does this only make sense if `tz_convert` requires its input to be sorted?  I don't think it does.

Reply:
yeah not sure why that indexer is there

Reply:
In the next pass I'll see if this can be simplified.